dbcon/mysql/ha_mcs_datatype.h

```diff
@@ -188,6 +188,20 @@ class StoreFieldMariaDB: public StoreField
 
 class WriteBatchFieldMariaDB: public WriteBatchField
 {
+  // Maximum number of decimal digits that can be represented in 4 bytes
+  static const int DIG_PER_DEC = 9;
+  // See strings/decimal.c
+  const int dig2bytes[DIG_PER_DEC+1]={0, 1, 1, 2, 2, 3, 3, 4, 4, 4};
+
+
+  // Returns the number of bytes required to store a given number
+  // of decimal digits
+  int numDecimalBytes(int digits)
+  {
+    return (((digits/DIG_PER_DEC) * 4) + dig2bytes[digits % DIG_PER_DEC]);
```

Comment:
I might had loost the point but my math tells me this f() returns 17 bytes for DECIMAL(38). Did I get it right that this f() works for precision <= 18?

Reply:
That's correct, the server sends 17 bytes in the binary buffer if `digits == 38`. Yes, this function also works correctly for precision <= 18. I have verified that for every possible combination of precision and scale, the bytes computed with the old logic and the new simplified logic are the same.